BZip + no gcc compiler on osx machines fails.

Unless I explicitly specify my CC and CXX env variables to point to clang, the bzip2 build tries to use gcc. I would think that it should default to cc / c++ if they exist over gcc/g++

```
Created build dir: /Users/robert/Work/hashdist/build/tmp/bzip2-mwq3d3ddgkuw
[bzip2] Building bzip2/mwq3d3ddgkuw
Unpacking sources files:svausq7ipn6lvz3iogmygwgqs2cy2tto
Unpacking sources tar.gz:ukci6nh42xlm6r666acgd7fvfcqeqtmo
[bzip2] running [u'/bin/bash', '_hashdist/build.sh']
[bzip2] environment:
[bzip2]   {'ARTIFACT': u'/Users/robert/Work/hashdist/build/bld/bzip2/mwq3d3ddgkuw',
[bzip2]    'BASH': u'/bin/bash',
[bzip2]    'BUILD': u'/Users/robert/Work/hashdist/build/tmp/bzip2-mwq3d3ddgkuw',
[bzip2]    'HASHDIST_CPU_COUNT': '1',
[bzip2]    'HDIST_CONFIG': '{"gc_roots":"/Users/robert/Work/hashdist/build/gcroots","build_stores":[{"dir":"/Users/robert/Work/hashdist/build/bld"}],"source_caches":[{"dir":"/Users/robert/Work/hashdist/build/src"}],"cache":"/Users/robert/Work/hashdist/build/cache","build_temp":"/Users/robert/Work/hashdist/build/tmp"}',
[bzip2]    'HDIST_IMPORT': '',
[bzip2]    'HDIST_IMPORT_PATHS': '',
[bzip2]    'HDIST_VIRTUALS': '',
[bzip2]    'PATH': u'/usr/local/bin:/usr/bin:/bin',
[bzip2]    'PWD': u'/Users/robert/Work/hashdist/build/tmp/bzip2-mwq3d3ddgkuw'}
[bzip2] 
[bzip2] If compilation produces errors, or a large number of warnings,
[bzip2] please read README.COMPILATION.PROBLEMS -- you might be able to
[bzip2] adjust the flags in this Makefile to improve matters.
[bzip2] 
[bzip2] Also in README.COMPILATION.PROBLEMS are some hints that may help
[bzip2] if your build produces an executable which is unable to correctly
[bzip2] handle so-called 'large files' -- files of size 2GB or more.
[bzip2] 
[bzip2] make: gcc: No such file or directory
[bzip2] gcc -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64 -c blocksort.c
[bzip2] make: *** [blocksort.o] Error 1
[bzip2 ERROR] Command '[u'/bin/bash', '_hashdist/build.sh']' returned non-zero exit status 2
[bzip2 ERROR] command failed (code=2); raising
```

The system.yaml that was used to create this issue looked like:

```
extends:
 - file: osx.yaml

  PROLOGUE: |
    export MACOSX_DEPLOYMENT_TARGET=$(sw_vers -productVersion | sed "s/\(10.[0-9]\).*/\1/");

#export CXX=/usr/bin/clang++; export CC=/usr/bin/clang;

packages:
  cmake:
    use: host-cmake
  mpi:
    use: host-mpi
 bzip2:
  zlib:
  szip:
  png:
  libxml2:
  hdf5:

```

--- a/pkgs/bzip2/bzip2.yaml
+++ b/pkgs/bzip2/bzip2.yaml
@@ -10,13 +10,20 @@ sources:
 
 build_stages:
 
+- name: patch
+  before: make-configure
+  files: [clang-fix.patch]
+  handler: bash
+  bash: |
+    patch -up1 < _hashdist/clang-fix.patch
+
 - name: make-configure
   after: prologue
   handler: bash
-  when platform == 'linux':  
+  when platform == 'linux':
     bash: |
       make -f Makefile-libbz2_so
-  when platform == 'Darwin':  
+  when platform == 'Darwin':
     bash: |
       make -f Makefile
 

--- a/pkgs/bzip2/clang-fix.patch
+++ b/pkgs/bzip2/clang-fix.patch
@@ -1,0 +1,10 @@
+--- bzip2-1.0.6.orig/Makefile 2014-03-12 12:42:45.000000000 -0400
++++ bzip2-1.0.6/Makefile  2014-03-12 12:54:03.000000000 -0400
+@@ -15,7 +15,6 @@
+ SHELL=/bin/sh
+
+ # To assist in cross-compiling
+-CC=gcc
+ AR=ar
+ RANLIB=ranlib
+ LDFLAGS=

